File: core/shared/src/main/scala/diplomacy/rule/Order.scala
package diplomacy.rule

import diplomacy.board.{ Power, MilitaryBranch, DiplomacyUnit }

trait Order[Power_ <: Power, MilitaryBranch_ <: MilitaryBranch] {
  val unit: DiplomacyUnit[Power_, MilitaryBranch_]
}

File: lib/rule/order.js
'use strict'

/**
 * @classdesc Order of DiplomacyMap
 * @memberof rule
 * @prop {!board.Unit} unit - The unit of Diplomacy that corresponds to this order.
 */
class Order {
  /**
   * @constructor
   * @param {!board.Unit} unit - The unit of Diplomacy that corresponds to this order.
   */
  constructor (unit) {
    this.unit = unit
  }
}

module.exports = Order
